invenio_records_resources/services/records/service.py

```diff
@@ -579,3 +581,85 @@ def on_relation_update(
 
             self.reindex(identity, search_query=search_query)
         return True
+
+    @unit_of_work()
+    def create_or_update_many(self, identity, data, uow=None):
+        """Create or update a list of records.
+
+        This method takes a list of record data and creates or updates the corresponding records.
+
+        :param identity: The user identity performing the operation.
+        :param data: A list of tuples containing the record ID and record data.
+        :param uow: The unit of work to register the record operations. Defaults to None.
+        """
+        records_processed = []
+
+        def _create_record(record_dict, record):
+            """Create a record."""
+            # Permissions
+            self.require_permission(identity, "update", record=record)
+            record_data, schema_errors = self.schema.load(
+                record_dict,
+                context=dict(identity=identity, pid=record.pid, record=record),
+                raise_errors=False,
+            )
+
+            # If errors we avoid creating the record
+            if schema_errors:
+                records_processed.append(("update", record_dict, schema_errors, None))
+                return
+
+            # Run components
+            self.run_components(
+                "update", identity, data=record_data, record=record, uow=uow
+            )
+
+            records_processed.append(("update", record, schema_errors, None))
+
+        def _update_record(record_dict):
+            """Update a record."""
+            self.require_permission(identity, "create")
+
+            # Validate data and create record with pid
+            record_data, schema_errors = self.schema.load(
+                record_dict, context={"identity": identity}, raise_errors=False
+            )
+
+            # If errors we avoid creating the record
+            if schema_errors:
+                records_processed.append(("create", record_dict, schema_errors, None))
+                return
+
+            # It's the components who saves the actual data in the record.
+            record = self.record_cls.create({})
+
+            # Run components
+            self.run_components(
+                "create",
+                identity,
+                data=record_data,
+                record=record,
+                errors=schema_errors,
+                uow=uow,
+            )
+            records_processed.append(("create", record, schema_errors, None))
+
+        # We avoid using create and update methods to bulk index all records at once
+        for record_id, record_dict in data:
+            try:
+                record = self.record_cls.pid.resolve(record_id)
+                _create_record(record_dict, record)
+            except (NoResultFound, PIDDoesNotExistError):
+                _update_record(record_dict)
```

Comment:
```suggestion
                _update_record(record_dict, record)
            except (NoResultFound, PIDDoesNotExistError):
                _create_record(record_dict)
```

I think these should be swapped?

Reply:
.... Right! Thanks for realising!